### ISSUE TITLE
Unexpected TransportOperation RequiredDispatchConsistency value

### ISSUE THREAD

**Issue by mauroservienti** [Particular/NServiceBus#5763]
The following code generates a `TransportOperation` with a `RequiredDispatchConsistency`  value of `Default`

```
var messageSession = ...
messageSession.Send(new Message());
```

Even though I was expecting the `RequiredDispatchConsistency` to be `Immediate`, as it's what effectively the operation is: it's an immediate operation. Is this intended? If it is, what's the use case for immediate operations in the above scenario? Like the following?

```
var messageSession = ...
var options = new SendOptions();
options.RequireImmediateDispatch();
messageSession.Send(new Message(), options);
```

**Comment by andreasohlund** [comment]
Default means "participate in the incoming transaction if possible" while `RequireImmediateDispatch` means "push it straight to the transport with no batching or transaction".

Is it the fact that you call it from outside a handler that causes you to feel that it should be immediate?

**Comment by mauroservienti** [comment]
Correct, I'm using the message session, not the message context, and I'm not in the context of an incoming message, I was expecting it to be immediate.

**Comment by andreasohlund** [comment]
That would mean that transaction scope aware transports like sqlserver and msmq would not participate in transaction scopes when used outside a handler, not sure that is what one would expect?

**Comment by mauroservienti** [comment]
Does that even work or does it require way much more code than a simple:

```csharp
using var scope = new TransactionScope(...);
messageSession.Send(...);
scope.Complete();
```

the above pseudo code snipetts, afaik, doesn't work. And it requires way much more code and knowledge of internals.

**Comment by andreasohlund** [comment]
Needs to be verified but I think msmq will enlist since when there is[ no transport transaction around it will do a send](https://github.com/Particular/NServiceBus.Transport.Msmq/blob/master/src/NServiceBus.Transport.Msmq/MsmqMessageDispatcher.cs#L107) with ["Automatic" if there is an ambient scope](https://github.com/Particular/NServiceBus.Transport.Msmq/blob/master/src/NServiceBus.Transport.Msmq/MsmqMessageDispatcher.cs#L199) around.

Would you be able to confirm?

**Comment by mauroservienti** [comment]
MSMQ is legacy, I think it makes more sense to look at SqlTransport

**Comment by andreasohlund** [comment]
Looks like it enlists in ambient transactions as well https://github.com/Particular/NServiceBus.SqlServer/blob/master/src/NServiceBus.Transport.SqlServer/Sending/MessageDispatcher.cs#L126
